File: algorithms/julia/fibonacci-dp.jl
const MOD = 1_000_000_007

function fib_dp(n::Int)
    memo = zeros(Int, n + 1)
    n >= 1 && (memo[2] = 1)
    for i in 3:n+1
        memo[i] = (memo[i-1] + memo[i-2]) % MOD
    end
    return memo[n+1]
end

function main()
    n = parse(Int, strip(read(stdin, String)))

    fib_dp(10)  # JIT warmup

    start  = time_ns()
    result = fib_dp(n)
    ms     = (time_ns() - start) / 1e6

    println(result)
    println(stderr, ms)
end

main()

File: algorithms/rust/fibonacci-dp.rs
use std::io::{self, Read, Write, BufWriter};
use std::time::Instant;

const MOD: u64 = 1_000_000_007;

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    let start = Instant::now();

    let mut memo = vec![0u64; n + 1];
    if n > 0 { memo[1] = 1; }
    for i in 2..=n {
        memo[i] = (memo[i-1] + memo[i-2]) % MOD;
    }

    let ms = start.elapsed().as_secs_f64() * 1000.0;

    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    writeln!(out, "{}", memo[n]).unwrap();
    eprintln!("{:.3}", ms);
}